### PR TITLE
Added rolling hashes to JSON Outputter

### PR DESCRIPTION
**Pull request by kirkeaton**
1. Added Rolling hashes to JSON Outputter
2. Renamed JDAO into AbstractJDAO
3. Created HashingJDAO
4. Created RollingHashingJDAO

### CODE REVIEW

File: src/foam/crypto/hash/HashingJDAO.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright 2017,2018 The FOAM Authors. All Rights Reserved.
```
Comment (kgrgreer) [inline src/foam/crypto/hash/HashingJDAO.java]: Copyright should just be the first year of creation (according to Google lawyers) and doesn't need to ever be updated after that.

File: src/foam/crypto/hash/HashingJDAO.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+import foam.core.ClassInfo;
+import foam.core.X;
+import foam.dao.AbstractJDAO;
+import foam.dao.DAO;
+import foam.dao.MapDAO;
+import foam.lib.json.Outputter;
+import foam.lib.json.OutputterMode;
+
+public class HashingJDAO
```
Comment (kgrgreer) [inline src/foam/crypto/hash/HashingJDAO.java]: Is this cascading/chained?

File: src/foam/crypto/hash/HashingJDAO.java (hunk 1 above)
Comment (kirkeaton) [reply src/foam/crypto/hash/HashingJDAO.java]: Fixed

File: src/foam/dao/AbstractJDAO.java [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+import java.io.*;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.regex.Pattern;
+import java.util.TimeZone;
+
+public abstract class AbstractJDAO
+    extends ProxyDAO
+{
+  protected Pattern COMMENT = Pattern.compile("(/\\*([^*]|[\\r\\n]|(\\*+([^*/]|[\\r\\n])))*\\*+/)|(//.*)");
```
Comment (kgrgreer) [inline src/foam/dao/AbstractJDAO.java:28]: This seems like overkill.  Isn't it enough to just check if the first two characters are //?
Not really related to this PR though.

Comment (kirkeaton) [reply src/foam/dao/AbstractJDAO.java:28]: Kind of is yeah but I wanted it to be exhaustive

**Pull request merged**